src/ai/ai.ts

```diff
@@ -0,0 +1,172 @@
+import { OperationType, getService, request } from '../ainize';
+import {
+  AiConfigurationTransactionResult,
+  CreditTransactionResult,
+  NftToken,
+  NftTokens,
+  QueryParams,
+  TokenStatus,
+} from '../types';
+import { buildSetTxBody, buildSetValueOp, sleep, sendTx } from '../utils/util';
+import { Path } from '../utils/path';
+import { validateObject, validateObjectOwner } from '../utils/validator';
+import { DEFAULT_AINIZE_SERVICE_NAME } from '../constants';
+
+/**
+ * This class manages ai configurations for AINFT object,\
+ * manages the credits needed for their uses, and checks if tokens are available for creation.\
+ * Do not create it directly; Get it from AinftJs instance.
+ */
+export class Ai extends FactoryBase {
+  /**
+   * Sets up ai configuration for an AINFT object.
+   * @param {string} objectId - The ID of the AINFT object.
+   * @param {string} serviceName - The name of Ainize service.
+   * @returns {Promise<AiConfigurationTransactionResult>} Returns a promise that resolves with both the transaction result and the configuration info.
+   */
+  async configure(
+    objectId: string,
+    serviceName: string
+  ): Promise<AiConfigurationTransactionResult> {
+    const address = await this.ain.signer.getAddress();
+
+    await validateObject(this.ain, objectId);
+    await validateObjectOwner(this.ain, objectId, address);
+    await getService(this.ainize!, serviceName); // NOTE(jiyoung): check if the service is deployed on Ainize.
+
+    const txBody = this.buildTxBodyForConfigureAi(objectId, serviceName, address);
+    const result = await sendTx(this.ain, txBody);
+
+    return { ...result, config: { name: serviceName } };
+  }
+
+  /**
+   * Retrieves the credit balance for a service.
+   * @param {string} serviceName - The name of Ainize service.
+   * @returns {Promise<number|null>} Returns a promise that resolves with the credit balance.
+   */
+  async getCredit(serviceName: string): Promise<number> {
+    const address = await this.ain.signer.getAddress();
+
+    let balance = 0;
+
+    if (serviceName === DEFAULT_AINIZE_SERVICE_NAME) {
+      const opType = OperationType.GET_CREDIT;
+      const body = { address };
+
+      const response = await request<number>(this.ainize!, {
+        serviceName: serviceName,
+        opType,
+        data: body,
+        timeout: 2*60*1000, // 2min
+      });
+      balance = response.data;
+    } else {
+      const service = await getService(this.ainize!, serviceName);
+      balance = await service.getCreditBalance();
+    }
+
+    return balance;
+  }
+
+  /**
+   * @todo Enable deposit function once withdrawal is fully implemented and tested.
+   * Deposits a credits for a service.
+   * @param {string} serviceName - The name of Ainize service.
+   * @param {number} amount - The amount of credits to deposit.
+   * @returns {Promise<CreditTransactionResult>} Returns a promise that resolves with the deposit transaction info.
+   */
+  /*
```

Comment:
could you add a todo describing this commenting or when it will be uncommented?

Reply:
I added @todo field in the comments. ([line:78])(https://github.com/ainft-team/ainft-js/pull/286/files/7de43d78aa0cbb1e1016b78feacf2168f647710c#diff-ac1856ed663bd0d5ea0185f6587d50114a1ef051bccc90f663627ca8fb9ed9d8R78)